eth/tracers/api_test.go

```diff
@@ -601,6 +606,85 @@ func TestTraceBlock(t *testing.T) {
+
+	// Initialize test accounts
+	accounts := newAccounts(2)
+	genesis := &core.Genesis{
+		Config: params.IstanbulEHFTestChainConfig,
+		Alloc: core.GenesisAlloc{
+			accounts[0].addr: {Balance: big.NewInt(231001)},
+			accounts[1].addr: {Balance: common.Big0},
+			common.HexToAddress("0xce10"): { // Registry Proxy
+				Code: RegistryProxyOpcodes,
+				Storage: map[common.Hash]common.Hash{
+					common.HexToHash("0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc"): common.HexToHash("0xce11"), // Registry Implementation
+					common.HexToHash("0x91646b8507bf2e54d7c3de9155442ba111546b81af1cbdd1f68eeb6926b98d58"): common.HexToHash("0xd023"), // Governance Proxy
+				},
+				Balance: big.NewInt(0),
+			},
+			common.HexToAddress("0xce11"): { // Registry Implementation
+				Code:    RegistryOpcodes,
+				Balance: big.NewInt(0),
+			},
+		},
+	}
+
+	api := NewAPI(newTestBackend(t, 1, genesis, func(i int, b *core.BlockGen) {
+		// The block base fee is mocked to be 3
+		// Two transactions are build, so that the correct gas price is 5 (base fee of 3 + tip of 2), but the
+		// incorrect calculation leads to a gas price of 6.
+		// The account balance is chosen in a way that the second transaction won't be able to execute if the
+		// calculation is wrong.
+
+		tip := big.NewInt(2)
+		cap := big.NewInt(6)
+
+		txdata1 := types.NewTx(&types.CeloDynamicFeeTx{
+			ChainID:   params.IstanbulEHFTestChainConfig.ChainID,
```

Comment:
As far as I understand, the core.BlockGen, it has the config. So you could access the same config you used to create it, which will avoid future updates if we change some of those chainConfigs

Reply:
Good point, I added a getter and accessed it from there.